src/geohash_helper.c

```diff
@@ -101,11 +101,72 @@ uint8_t geohashEstimateStepsByRadius(double range_meters, double lat) {
  */
 int geohashBoundingBox(GeoShape *shape, double *bounds) {
     if (!bounds) return 0;
+    double height = 0.0, width = 0.0;
+    if (shape->type == CIRCULAR_TYPE) {
+        // For circular shapes, use the given radius directly.
+        height = shape->conversion * shape->t.radius;
+        width = shape->conversion * shape->t.radius;
+    } else if (shape->type == RECTANGLE_TYPE) {
+        // For rectangles, calculate the diagonal as the radius.
+        height = shape->conversion * shape->t.r.height / 2;
+        width = shape->conversion * shape->t.r.width / 2;
+    } else if (shape->type == POLYGON_TYPE) {
```

Comment:
Currently, this block of code converts Lon & Lat to Cartesian Coordinates to find the Bounding Box **and** Centroid of the polygon.

I am considering the following:
* Use Lon and Lat to find the Bounding Box
* Convert to Cartesian Coordinates to compute the Centroid